src/MLOps.NET/MLOpsContext.cs

```diff
@@ -1,107 +1,34 @@
-﻿using Dynamitey;
-using MLOps.NET.Entities.Entities;
+﻿using MLOps.NET.Catalogs;
 using MLOps.NET.Storage;
 using System;
-using System.Linq;
-using System.Reflection;
-using System.Threading.Tasks;
 
 namespace MLOps.NET
 {
     ///<inheritdoc cref="IMLOpsContext"/>
     public class MLOpsContext : IMLOpsContext
     {
-        private readonly IMetaDataStore metaDataStore;
-        private readonly IModelRepository modelRepository;
-
         internal MLOpsContext(IMetaDataStore metaDataStore, IModelRepository modelRepository)
         {
-            this.metaDataStore = metaDataStore ?? throw new ArgumentNullException(nameof(metaDataStore));
-            this.modelRepository = modelRepository ?? throw new ArgumentNullException(nameof(modelRepository));
-        }
+            if (metaDataStore == null) throw new ArgumentNullException(nameof(metaDataStore));
+            if (modelRepository == null) throw new ArgumentNullException(nameof(modelRepository));
 
-        ///<inheritdoc/>
-        public async Task<Guid> CreateExperimentAsync(string name)
-        {
-            return await metaDataStore.CreateExperimentAsync(name);
+            this.LifeCycleCatalog = new LifeCycleCatalog(metaDataStore);
+            this.Evaluation = new EvaluationCatalog(metaDataStore);
+            this.Model = new ModelCatalog(modelRepository);
+            this.Training = new TrainingCatalog(metaDataStore);
         }
 
-        ///<inheritdoc/>
-        public async Task<Guid> CreateRunAsync(Guid experimentId)
-        {
-            return await metaDataStore.CreateRunAsync(experimentId);
-        }
+        ///<inheritdoc cref="IMLOpsContext"/>
+        public LifeCycleCatalog LifeCycleCatalog { get; set; }
```

Comment:
Do we want to expose setters also?

Reply:
Great catch @shanranm, I did not intend to do that. Pushed a change.